homeassistant/components/tahoma/cover.py

```diff
@@ -109,12 +112,18 @@ def update(self):
         #   _position: 0 is closed, 100 is fully open.
         #   'core:ClosureState': 100 is closed, 0 is fully open.
         if self._closure is not None:
-            self._position = 100 - self._closure
+            if self.tahoma_device.type == 'io:HorizontalAwningIOComponent':
```

Comment:
can this be `if self.device_class == DEVICE_CLASS_AWNING`?

Reply:
There's also a "Vertical Awning" supported by the platform - I believe that one's more like a shade